### ISSUE TITLE
Documentation: clarify that callback() may return before callback chain is complete

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4439]
|<img alt="jrennie's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| jrennie reported|
|-|-|
|Trac ID|trac#4439|
|Type|enhancement|
|Created|2010-05-17 19:49:41Z|
|Branch|https://github.com/twisted/twisted/tree/callback-chaining-doc-4439|

This ticket simply requests that documentation for the callback and errback methods of twisted.internet.defer.Deferred clarify when they will return.  As currently written, a programmer might assume that these methods will not return until the chain has completed.  In fact, the methods will return as soon as they run into a Deferred-returning function in the chain.  'course, it's probably best to provide no guarantees as to when these functions will return in case the Twisted implementation changes.

<details><summary>Searchable metadata</summary>

```
trac-id__4439 4439
type__enhancement enhancement
reporter__jrennie jrennie
priority__normal normal
milestone__ 
branch__branches_callback_chaining_doc_4439 branches/callback-chaining-doc-4439
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1274125781000000 1274125781000000
changetime__1275269773000000 1275269773000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [29049]) Branching to 'callback-chaining-doc-4439'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [29050]) Expand `Deferred.callback` docstring

refs [#4439](https://github.com/twisted/twisted/issues/4439)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|ivank commented|
|-|-|

Very minor nit: `*error*- callback` should be `*error*-callback`. Everything else looks good, please merge.

**Comment by twisted-trac** [comment]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ivank set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [29113]) Merge callback-chaining-doc-4439

Author: exarkun
Reviewer: ivank
Fixes: [#4439](https://github.com/twisted/twisted/issues/4439)

Augment the `Deferred.callback` docstring with text about chaining.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
